applications/train.py

```diff
@@ -160,6 +174,41 @@ def validate(model, dataloader, criterion, metrics, second_metrics, nc):
         
         # predict the model output
         outputs = model(img_noisy)
+        
+        # save figs for later reference
+        # selecting random sample in batch to visualize
+        a_inp = img_noisy.cpu().detach().numpy()[:,nc-1,:,:]
+        print(int(a_inp.shape[0]))
+        samp_ = np.random.choice(np.arange(0,int(a_inp.shape[0]),1))
+
+        # change plt logging level otherwise get a lot of debug output
+        plt.set_loglevel(level='warning')
+
+        # input
+        im = plt.imshow(a_inp[samp_])
+        plt.colorbar(im)
+        plt.savefig(
+            f"{save_loc}/trial{str(trial_num)}/{data_split}_inp_{str(epoch)}_{str(trial_num)}.png", 
+            bbox_inches='tight')
+        plt.close()
+
+        # output
+        b_out = outputs.cpu().detach().numpy()[:,0,:,:][samp_]
+        im = plt.imshow(b_out)
+        plt.colorbar(im)
+        plt.savefig(
+            f"{save_loc}/trial{str(trial_num)}/{data_split}_out_{str(epoch)}_{str(trial_num)}.png", 
+            bbox_inches='tight')
+        plt.close()
+
+        # label
+        c_lbl = img_label.cpu().detach().numpy()[:,0,:,:][samp_]
+        im = plt.imshow(c_lbl)
+        plt.colorbar(im)
+        plt.savefig(
+            f"{save_loc}/trial{str(trial_num)}/{data_split}_lbl_{str(epoch)}_{str(trial_num)}.png", 
+            bbox_inches='tight')
+        plt.close()
```

Comment:
Looks great! I wonder if it would be better to make this section optional, could depend on one argument. Just in case we'd like to run validations without plotting anything.

Reply:
good point -- I am also getting annoyed with how many images are getting saved each epoch for EACH trial lol... I think every fifth epoch or something may be better